cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -1025,12 +1032,22 @@ private boolean isProgrammaticConfig(Configuration conf, String name) {
   @Nullable
   private Location createPluginArchive(Location targetDir) throws IOException {
     File pluginArchive = context.getPluginArchive();
-    if (pluginArchive == null) {
+    return copyFileToLocation(pluginArchive, targetDir);
+  }
+
+  /**
+   * Copies a file to the target location.
+   *
+   * @param targetDir directory where the file should be copied to.
+   * @return {@link Location} to the file or {@code null} if given file is {@code null}.
+   */
+  private Location copyFileToLocation(@Nullable File file, Location targetDir) throws IOException {
```

Comment:
So the return value can be null? If that's the case, annotate this method with `@Nullable`.